runtime/parser/declaration_test.go

```diff
@@ -3731,6 +3731,144 @@ func TestParseTransactionDeclaration(t *testing.T) {
 			result.Declarations(),
 		)
 	})
+
+	t.Run("role", func(t *testing.T) {
```

Comment:
maybe also add some tests for role blocks without prepare, empty role blocks, etc.

Reply:
Yea; would like to see tests for the failure cases here. In addition to what Supun mentioned above, would be good to have tests for nested role blocks, role blocks with conditions, and multiple role blocks in a transaction.

Reply:
Added more tests in 67105eb570ec74e27c045912b4aa3ea780a4e179.

Nesting and pre/post conditions are not supported